components/ShareBike.tsx

```diff
@@ -0,0 +1,150 @@
+import { Bike } from '../lib/bike'
+import { Button } from './Button'
+import { useState, FormEvent } from 'react'
+import { FormError } from './Form'
+import { Api } from '../lib/api'
+
+export function ShareBike({bike, api}: {bike: Bike, api: Api}) {
+    const [successModal, setSuccessModal] = useState(false)
+    const [error, setError] = useState<string | undefined>(undefined)
+    const [shareinfo, setShareinfo] = useState({
+        email: "",
+        bikeId: bike.id,
+        role: "user",
+        duration: 86400 // 86400 = 1 day but in seconds.
+    })
+
+    const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        try {
+            const result = await api.createBikeSharingInvitation(shareinfo)
+            if(result.result) {
+                // TODO: success popup
```

Comment:
A popup would indeed be nice but to much for this pr

Tough i think a user should at least get some feedback the bike has been shared..

I think we should add a state value `success` (`boolean`) that is set to true if it succeeded and if it succeeded show a message similar to the error with a success message